crypto/box/src/traits.rs

```diff
@@ -87,7 +92,11 @@ pub trait CryptoBox<KexAlgo: Kex>: Default {
     /// If status is false then mac check failed and plaintext should be zeroed.
     ///
     /// Meant to mirror aead::decrypt
-    fn decrypt(&self, key: &KexAlgo::Private, cryptogram: &[u8]) -> Result<(bool, Vec<u8>), Error> {
+    fn decrypt(
+        &self,
+        key: &KexAlgo::Private,
+        cryptogram: &[u8],
+    ) -> Result<(CtDecryptResult, Vec<u8>), Error> {
```

Comment:
there's a good argument that this should be made to return `Zeroizing<Vec<u8>>`, that would mean that all the callers in SDKs etc. dont need to explicitly zeroize this

Reply:
this passed deploy, and i decided to punt on zeroize for now because i can't find a good solution for zeroizing generic array. I posted a pull request here: 

https://github.com/fizyk20/generic-array/pull/112